tests/Sharpliner.Tests/AzureDevOps/StrongTypedTests.cs

```diff
@@ -35,7 +35,7 @@ private class Pipeline_With_Strong_Variables_And_Parameters : SimpleTestPipeline
                         If.Equal(parameters["Parameter1"], "SomeParameterValue1")
                             .Step(Script.Inline("echo Hello")),
 
-                        If.Equal(variables["VariableBasedUponParameter"], "Parameter1 Equals SomeParameterValue1")
+                        If.Equal(staticVariables["VariableBasedUponParameter"], "Parameter1 Equals SomeParameterValue1")
```

Comment:
Yeah, but then you have to already sort of know that a variable "Foo" is a runtime/static one. What I am trying to get at is that if I don't really know about how this works, I will:
1. try to use `variables[]` somewhere,
2. get a compilation error
3. read the docs/code
4. hopefully find that there's also `staticVariables`,
5. change `variables` to `staticVariables`,
6. compile YAML, try to run it and fail.

Technically, in step 5 I can realize I am using the wrong kind and that will give me a hint. In step 4 there's a room for failure where the user might become angry as it might be hard to figure out there's something else than `variables` (we would put it in the docs but it might be hard to discover this API though).

So in my view, the knowledge about which variable is runtime and which is compile time is already required for the end user because you need to know which variable you can use where and thus I think we don't really need 2 APIs. I would keep `variables` only as having two APIs still doesn't prevent me from trying to use a variable in the wrong context (I can still try to use a runtime variable name with the `staticVariables` API). The API does the nice thing for me that wherever I use `variables`, it will compile the right reference format for me in the YAML which is already super cool and useful.

But I am open to changing my mind if I missed an angle here.

Reply:
@thomhurst if you don't mind, I took the liberty of removing `staticVariables` and kept `variables` only. I think this way, things would still work quite nicely!

Reply:
Sorry I meant to get round to this and forgot!
Yeah, I think that's fine and removes some confusion.

Looks like your API test still has a reference to static variables though.

Reply:
Ah, will have a look

Reply:
@thomhurst I kept the `StaticVariableReference`, I only changed the `OneOf` to accept `VariableReference` instead. It felt like we can keep the work you've done in the code base (distinguishing what needs to go where). Hopefully I didn't allow for something invalid this way.

Reply:
Ok, I removed `StaticVariableReference`